doc/api/tls.md

```diff
@@ -876,6 +876,27 @@ socket.on('end', () => {
 });
 ```
 
+When using an instance of `net.Socket`, use `net.Socket` to upgrade an existing socket. Do not wrap the `net.Socket` in a `TLSSocket`. See the example below for usage of upgrading an existing socket:
```

Comment:
I think that's much more clear - the original issue had a user trying to do this:

`const tlsSocket = new TlsSocket(new NetSocket());`

And I think your wording will keep them from doing that much more clearly. Will update now!